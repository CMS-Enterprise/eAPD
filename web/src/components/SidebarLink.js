import PropTypes from 'prop-types';
import React from 'react';

const SidebarLink = ({ anchor, children, isActive }) => (
  <li className="mb-tiny">
    <a
      href={`#${anchor || '!'}`}
      className={`inline-block white text-decoration-none truncate ${
        isActive ? 'bold border-bottom border-width-3 border-blue' : ''
      }`}
    >
      {children}
    </a>
  </li>
);

SidebarLink.propTypes = {
  anchor: PropTypes.string,
  children: PropTypes.node.isRequired,
  isActive: PropTypes.bool
};

SidebarLink.defaultProps = {
  anchor: null,
  isActive: false
};

export default SidebarLink;
