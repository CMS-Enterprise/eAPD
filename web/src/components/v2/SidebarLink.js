import PropTypes from 'prop-types';
import React from 'react';

const SidebarLink = ({ children }) => (
  <li>
    <a href="#!" className="inline-block white text-decoration-none truncate">
      {children}
    </a>
  </li>
);

SidebarLink.propTypes = {
  children: PropTypes.node.isRequired
};

export default SidebarLink;
