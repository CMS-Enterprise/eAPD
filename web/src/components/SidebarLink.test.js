import { shallow } from 'enzyme';
import React from 'react';

import SidebarLink from './SidebarLink';

describe('SidebarLink component', () => {
  test('renders correctly in the default case', () => {
    const component = shallow(<SidebarLink>link text</SidebarLink>);
    expect(component).toMatchSnapshot();
  });

  test('renders link correctly if an anchor is given', () => {
    const component = shallow(
      <SidebarLink anchor="moop">link text</SidebarLink>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders correctly if link is active', () => {
    const component = shallow(
      <SidebarLink hash="moop" anchor="moop">
        link text
      </SidebarLink>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders correctly if link has sub-links', () => {
    const component = shallow(
      <SidebarLink sub={[{ id: 1, name: 'child' }]}>link text</SidebarLink>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders correctly if link is collapsed', () => {
    const component = shallow(
      <SidebarLink expanded={false} sub={[{ id: 1, name: 'child' }]}>
        link text
      </SidebarLink>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders correctly if link is expanded', () => {
    const component = shallow(
      <SidebarLink expanded sub={[{ id: 1, name: 'child' }]}>
        link text
      </SidebarLink>
    );
    expect(component).toMatchSnapshot();
  });
});
