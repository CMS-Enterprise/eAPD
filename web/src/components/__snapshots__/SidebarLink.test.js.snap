// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarLink component renders correctly if link is active 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarLink
    anchor={null}
    isActive={true}
  >
    link text
  </SidebarLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="inline-block white text-decoration-none truncate"
        href="#!"
      >
        link text
      </a>,
      "className": "mb-tiny relative sb-item-active",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "link text",
        "className": "inline-block white text-decoration-none truncate",
        "href": "#!",
      },
      "ref": null,
      "rendered": "link text",
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="inline-block white text-decoration-none truncate"
          href="#!"
        >
          link text
        </a>,
        "className": "mb-tiny relative sb-item-active",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "link text",
          "className": "inline-block white text-decoration-none truncate",
          "href": "#!",
        },
        "ref": null,
        "rendered": "link text",
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SidebarLink component renders correctly in the default case 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarLink
    anchor={null}
    isActive={false}
  >
    link text
  </SidebarLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="inline-block white text-decoration-none truncate"
        href="#!"
      >
        link text
      </a>,
      "className": "mb-tiny relative ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "link text",
        "className": "inline-block white text-decoration-none truncate",
        "href": "#!",
      },
      "ref": null,
      "rendered": "link text",
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="inline-block white text-decoration-none truncate"
          href="#!"
        >
          link text
        </a>,
        "className": "mb-tiny relative ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "link text",
          "className": "inline-block white text-decoration-none truncate",
          "href": "#!",
        },
        "ref": null,
        "rendered": "link text",
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SidebarLink component renders link correctly if an anchor is given 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarLink
    anchor="moop"
    isActive={false}
  >
    link text
  </SidebarLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="inline-block white text-decoration-none truncate"
        href="#moop"
      >
        link text
      </a>,
      "className": "mb-tiny relative ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "link text",
        "className": "inline-block white text-decoration-none truncate",
        "href": "#moop",
      },
      "ref": null,
      "rendered": "link text",
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="inline-block white text-decoration-none truncate"
          href="#moop"
        >
          link text
        </a>,
        "className": "mb-tiny relative ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "link text",
          "className": "inline-block white text-decoration-none truncate",
          "href": "#moop",
        },
        "ref": null,
        "rendered": "link text",
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
